pkg/filter/network/streamproxy/streamproxy.go

```diff
@@ -237,14 +239,49 @@ func (p *proxy) onUpstreamEvent(event api.ConnectionEvent) {
 	case api.ConnectFailed:
 		p.requestInfo.SetResponseFlag(api.UpstreamConnectionFailure)
 	}
+
+	p.onUpstreamEventStats(event, event.IsClose() || event == api.ConnectTimeout)
```

Comment:
we don't need to call `UpstreamConnectionActive.Dec` when the event is `api.ConnectTimeout`.